tools/cell_census_builder/__main__.py

```diff
@@ -120,8 +120,13 @@ def build(
     top_level_collection, filtered_datasets = build_step2_create_axis(
         soma_path, assets_path, datasets, experiment_builders, args
     )
-    assign_soma_joinids(filtered_datasets)
+
+    if not filtered_datasets:
+        logging.error("Zero datasets are suitable for processing. Aborting.")
+        return 1
```

Comment:
Fail here instead of later with an arcane, harder-to-troubleshoot error in code that doesn't expect empty set of datasets. Useful when using `--test-first-n`, which may result in 0 admissible datasets.